src/realm/object-store/sync/sync_session.cpp

```diff
@@ -381,10 +381,28 @@ std::function<void(util::Optional<app::AppError>)> SyncSession::handle_refresh(s
             }
         }
         else if (error) {
-            // 10 seconds is arbitrary, but it is to not swamp the server
-            std::this_thread::sleep_for(milliseconds(10000));
-            if (session_user) {
-                session_user->refresh_custom_data(handle_refresh(session));
+            if (error->http_status_code && *error->http_status_code == 401 &&
+                error->error_code == realm::app::make_error_code(realm::app::ServiceErrorCode::invalid_session)) {
```

Comment:
I don't think we need to check the error code here - if we get a 401, regardless of the error code, we should just give up. While we're here, we might as well handle 403 - I don't believe the server currently returns it, but it's in the same bucket of "if you get it, there's no point to retry" errors.